matchers/parse/token.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parse
+
+import (
+	"fmt"
+)
+
+type TokenKind int
+
+const (
+	TokenEOF TokenKind = iota
+	TokenOpenBrace
+	TokenCloseBrace
+	TokenComma
+	TokenEquals
+	TokenNotEquals
+	TokenMatches
+	TokenNotMatches
```

Comment:
```suggestion
	TokenOperatorEqual
	TokenOperatorNotEqual
	TokenOperatorRegExp
	TokenOperatorNotRegExp
```

To match the nomenclature on package labels.

Reply:
It's quite verbose, I'd much much rather change `pkg/labels` to what I've got here once we delete the old parser and move all the remaining code into the matchers package.

Reply:
If you don't want to rename it here - that's fine but I don't think I'm up for changing the others. We'll deal with the naming dissociation in both places then.